functors task (1)/student/include/impl.h

```diff
@@ -40,9 +45,16 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(AlcoholDrink alcoholDrink, NonAlcoholDrink nonAlcoholDrink)
     {
+        if (alcoholDrink == AlcoholDrink::Gin && nonAlcoholDrink == NonAlcoholDrink::LimeJuice)
+            return Cocktail::Gimlet;
+        if (alcoholDrink == AlcoholDrink::Gin && nonAlcoholDrink == NonAlcoholDrink::GrapefruitJuice)
```

Comment:
you can simplify these checks with switch

Reply:
Am I understand you right? Do you want to say that two nested switches (or switch with if's) are more readable than cascaded if's. Don't think so... both are ugly. In case of much bigger number of cocktails there should've been some kind of predefined array or map and just call find() here, but for 3 cocktails, I'm not sure this is really necessary.

Reply:
Yes, you're right about the map usage in this task (it's the best solution).
We can easily transform such kind of switch to a map (it will be discussed in "Containers" training).
Imho, switching logic is closer to mappng pattern, than such if, because you can see all pairs of values.